lib/TopStoriesFeed.jsm

```diff
@@ -85,10 +82,10 @@ this.TopStoriesFeed = class TopStoriesFeed {
     SectionsManager.onceInitialized(this.onInit.bind(this));
   }
 
-  observe(subject, topic, data) {
+  async observe(subject, topic, data) {
     switch (topic) {
       case "idle-daily":
-        this.updateDomainAffinityScores();
+        await this.updateDomainAffinityScores();
```

Comment:
With the typical parameter settings, have you done any benchmarks for this feature to get some basic idea about the typical required running time? Even better to do this on the reference machine. It's hard for me to review this part without those inputs.

The V1 personalization is currently being computed in the `idle-daily` handler since we are aware of its typical calculation duration is within hundreds of ms. Assuming V2 needs longer, computing it in daily-idle perhaps is not optimal anymore due to the fact that daily-idle is already crowded with other players such as daily maintenance for Places and IndexedDB quota manager.